spot_rl_experiments/spot_rl/utils/robot_subscriber.py

```diff
@@ -81,18 +86,35 @@ def __init__(self, spot=None, *args, **kwargs):
 
         rospy.loginfo(f"[{self.node_name}]: Robot subscribing has started.")
 
+    # Letting the user know about timeout to get images. This function lets the user know time taken to get the topics
+    def wait_for_images(self, subscriptions, timeout=30):
+        start_time = rospy.get_time()
+        while not all([self.msgs[s] is not None for s in subscriptions]):
+            if rospy.get_time() - start_time > timeout:
+                rospy.logwarn(f"[{self.node_name}]: Timeout waiting for image topics.")
+                break
```

Comment:
if we cannot get the images, do we want to continue running the code? Or we want to skill the program and let users know?

Reply:
This function gets timeout, tells the user that there's a timeout and continues execution for the topics that are publishing. It just replaces the old method , by giving a log of timeout for a topic.